Cache/Requests.lean

```diff
@@ -60,7 +60,7 @@ def downloadFile (hash : UInt64) : IO Bool := do
 
 /-- Calls `curl` to download files from the server to `CACHEDIR` (`.cache`) -/
 def downloadFiles (hashMap : IO.HashMap) (forceDownload : Bool) (parallel : Bool) : IO Unit := do
-  let hashMap := if forceDownload then hashMap else hashMap.filter (← IO.getLocalCacheSet) false
+  let hashMap := if forceDownload then hashMap else ← hashMap.filter false
```

Comment:
```suggestion
  let hashMap ← if forceDownload then pure hashMap else hashMap.filter false
```
you hit the footgun case of ← lifting over `if`

Reply:
Oof, can we have a linter?

Reply:
This should just be an error in core; most similar cases give you a "can't lift <- over binder" error. I think it's just an oversight.